File: src/types/expediente.ts
// Tipos para Expedientes
// Coinciden con los modelos de Rust

export type TipoExpediente = "InfoGov" | "Gde" | "Interno" | "Pago" | "Otro";
export type EstadoExpediente = "Iniciado" | "EnProceso" | "EnRevision" | "Observado" | "Finalizado" | "Archivado";
export type Prioridad = "Baja" | "Media" | "Alta" | "Urgente";

export interface Expediente {
  id: string;
  numero: string;
  año: number;
  tipo: TipoExpediente;
  
  // Campos comunes a todos los expedientes
  archivo?: string | null;
  nro_infogov?: string | null;
  tema?: string | null;
  nro_gde?: string | null;
  fecha_inicio: string;
  fecha_pase?: string | null;
  oficina?: string | null;
  estado: EstadoExpediente;
  buzon_grupal?: string | null;
  hacer?: string | null;
  resumen?: string | null; // Combinación de nro_infogov + resumen + nro_gde
  
  // Campos legacy/adicionales
  caratula?: string | null;
  resolucion_nro?: string | null;
  asunto: string;
  descripcion: string | null;
  area_responsable: string;
  prioridad: Prioridad;
  fecha_vencimiento: string | null;
  fecha_finalizacion: string | null;
  agente_responsable_id: string | null;
  archivos_adjuntos: string | null;
  observaciones: string | null;
  
  // Campos extra para expedientes de tipo "Pago"
  oc_señor?: string | null;
  oc_domicilio?: string | null;
  oc_cuit?: string | null;
  oc_descripcion_zona?: string | null;
  oc_forma_pago?: string | null;
  oc_plazo_entrega?: string | null;
  factura_path?: string | null;
  
  created_at: string;
  updated_at: string;
  synced_at: string | null;
}

export interface CreateExpedienteInput {
  numero: string;
  año: number;
  tipo: TipoExpediente;
  
  // Campos comunes
  archivo?: string;
  nro_infogov?: string;
  tema?: string;
  nro_gde?: string;
  fecha_inicio: string;
  fecha_pase?: string;
  oficina?: string;
  buzon_grupal?: string;
  hacer?: string;
  
  // Campos legacy/adicionales
  caratula?: string;
  resolucion_nro?: string;
  asunto: string;
  descripcion?: string;
  area_responsable: string;
  prioridad: Prioridad;
  fecha_vencimiento?: string;
  agente_responsable_id?: string;
  
  // Campos OC para expedientes de tipo "Pago"
  oc_señor?: string;
  oc_domicilio?: string;
  oc_cuit?: string;
  oc_descripcion_zona?: string;
  oc_forma_pago?: string;
  oc_plazo_entrega?: string;
  factura_path?: string;
}

export interface UpdateExpedienteInput {
  archivo?: string;
  nro_infogov?: string;
  tema?: string;
  nro_gde?: string;
  fecha_pase?: string;
  oficina?: string;
  buzon_grupal?: string;
  hacer?: string;
  asunto?: string;
  descripcion?: string;
  caratula?: string;
  resolucion_nro?: string;
  prioridad?: Prioridad;
  estado?: EstadoExpediente;
  fecha_vencimiento?: string;
  fecha_finalizacion?: string;
  agente_responsable_id?: string;
  observaciones?: string;
  
  // Campos OC para expedientes de tipo "Pago"
  oc_señor?: string;
  oc_domicilio?: string;
  oc_cuit?: string;
  oc_descripcion_zona?: string;
  oc_forma_pago?: string;
  oc_plazo_entrega?: string;
  factura_path?: string;
}

File: src-tauri/src/models/expediente.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

/// Modelo de Expediente
/// Representa un expediente administrativo (InfoGov, GDE, etc.)
#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Expediente {
    /// ID único (UUID v4 como string)
    pub id: String,
    
    /// Identificación del expediente
    pub numero: String,
    pub año: i32,
    pub tipo: TipoExpediente,

    /// Identificadores externos
    pub nro_infogov: Option<String>,
    pub nro_gde: Option<String>,
    pub caratula: Option<String>,
    pub resolucion_nro: Option<String>,
    
    /// Descripción y contenido
    pub asunto: String,
    pub descripcion: Option<String>,
    
    /// Categorización
    pub area_responsable: String,
    pub prioridad: Prioridad,
    pub estado: EstadoExpediente,
    
    /// Fechas importantes
    pub fecha_inicio: DateTime<Utc>,
    pub fecha_vencimiento: Option<DateTime<Utc>>,
    pub fecha_finalizacion: Option<DateTime<Utc>>,
    
    /// Vinculaciones
    pub agente_responsable_id: Option<String>,
    
    /// Archivos adjuntos (rutas relativas)
    pub archivos_adjuntos: Option<String>, // JSON array de rutas
    
    /// Observaciones
    pub observaciones: Option<String>,
    
    /// Campos para Órdenes de Compra (solo si es expediente de pago)
    /// Proveedor/Beneficiario
    pub oc_señor: Option<String>,
    pub oc_domicilio: Option<String>,
    pub oc_cuit: Option<String>,
    
    /// Descripción de zona
    pub oc_descripcion_zona: Option<String>,
    
    /// Datos de forma de pago y plazo
    pub oc_forma_pago: Option<String>,
    pub oc_plazo_entrega: Option<String>,
    
    /// Metadatos
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub synced_at: Option<DateTime<Utc>>,
}

/// Tipo de expediente según el sistema
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "tipo_expediente", rename_all = "SCREAMING_SNAKE_CASE")]
pub enum TipoExpediente {
    #[sqlx(rename = "INFOGOV")]
    InfoGov,
    #[sqlx(rename = "GDE")]
    Gde,
    #[sqlx(rename = "INTERNO")]
    Interno,
    #[sqlx(rename = "PAGO")]
    Pago,
    #[sqlx(rename = "OTRO")]
    Otro,
}

/// Estado del expediente en su ciclo de vida
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "estado_expediente", rename_all = "SCREAMING_SNAKE_CASE")]
pub enum EstadoExpediente {
    #[sqlx(rename = "INICIADO")]
    Iniciado,
    #[sqlx(rename = "ENPROCESO")]
    EnProceso,
    #[sqlx(rename = "ENREVISION")]
    EnRevision,
    #[sqlx(rename = "OBSERVADO")]
    Observado,
    #[sqlx(rename = "FINALIZADO")]
    Finalizado,
    #[sqlx(rename = "ARCHIVADO")]
    Archivado,
}

/// Nivel de prioridad
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "prioridad", rename_all = "SCREAMING_SNAKE_CASE")]
pub enum Prioridad {
    #[sqlx(rename = "BAJA")]
    Baja,
    #[sqlx(rename = "MEDIA")]
    Media,
    #[sqlx(rename = "ALTA")]
    Alta,
    #[sqlx(rename = "URGENTE")]
    Urgente,
}

/// Datos para crear un expediente
#[derive(Debug, Deserialize)]
pub struct CreateExpediente {
    pub numero: String,
    pub año: i32,
    pub tipo: TipoExpediente,
    pub nro_infogov: Option<String>,
    pub nro_gde: Option<String>,
    pub caratula: Option<String>,
    pub resolucion_nro: Option<String>,
    pub asunto: String,
    pub descripcion: Option<String>,
    pub area_responsable: String,
    pub prioridad: Prioridad,
    pub fecha_inicio: DateTime<Utc>,
    pub fecha_vencimiento: Option<DateTime<Utc>>,
    pub agente_responsable_id: Option<String>,
    // Campos específicos para expedientes de tipo "Pago"
    pub oc_señor: Option<String>,
    pub oc_domicilio: Option<String>,
    pub oc_cuit: Option<String>,
    pub oc_descripcion_zona: Option<String>,
    pub oc_forma_pago: Option<String>,
    pub oc_plazo_entrega: Option<String>,
}

/// Datos para actualizar un expediente
#[derive(Debug, Deserialize)]
pub struct UpdateExpediente {
    pub asunto: Option<String>,
    pub descripcion: Option<String>,
    pub nro_infogov: Option<String>,
    pub nro_gde: Option<String>,
    pub caratula: Option<String>,
    pub resolucion_nro: Option<String>,
    pub prioridad: Option<Prioridad>,
    pub estado: Option<EstadoExpediente>,
    pub fecha_vencimiento: Option<DateTime<Utc>>,
    pub fecha_finalizacion: Option<DateTime<Utc>>,
    pub agente_responsable_id: Option<String>,
    pub observaciones: Option<String>,
}
